### PR TITLE
Fixes #35857 - don't assume single_content_view etc. is present in RABL templates

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

In our RABL templates, `single_content_view` and `single_lifecycle_environment` were being relied on to be non-nil. This caused some NilClass errors when a host had a cleared-out content facet.

#### Considerations taken when implementing this change?

hopefully this will fix nightly

#### What are the testing steps for this pull request?

ermm.. run the pipeline I guess?

### CODE REVIEW

File: app/views/katello/api/v2/content_facet/show.json.rabl [hunk 1]
```diff
@@ -4,19 +4,21 @@ child :content_facet => :content_facet_attributes do
   node do |content_facet|
     content_facet.single_content_view # just so Rails gives a warning and we remember to fix this
     version = content_facet.content_view_environments.first.content_view_version
```
Comment (ekohl) [inline app/views/katello/api/v2/content_facet/show.json.rabl:6]: I'm guessing `content_facet.content_view_environments.first` can be `nil` too. Not sure if `content_facet.content_view_environments` is always a list of even that can be `nil`.
```suggestion
    version = content_facet.content_view_environments.first&.content_view_version
```

**Review by ekohl — COMMENTED** [review]
Is `nil.present?` a Rails enhancement? In plain Ruby that doesn't work. There you can just use `if nil`.

**Comment by jeremylenz** [comment]
Yes, `.present?` is from Rails. But I think that should be fine, right?

**Comment by ekohl** [comment]
Yes, I'm just not used to Rails' enhancements since I often write in vanilla Ruby.

File: app/views/katello/api/v2/content_facet/show.json.rabl [hunk 2]
```diff
@@ -3,20 +3,22 @@ child :content_facet => :content_facet_attributes do
...
+        :content_view_version => version.version,
+        :content_view_version_id => version.id,
+        :content_view_version_latest => version.latest?
+      }
+    end
   end
 
   node :content_view_default? do |content_facet|
-    content_facet.single_content_view.default?
+    !!content_facet.single_content_view&.default?
```
Comment (parthaa) [inline app/views/katello/api/v2/content_facet/show.json.rabl:17]: This seems a very javascripty thing and not ruby?
Why is this needed. The code prior to your single-cv change
`content_facet.content_view.default?` seems to think content facet has to have a CV

File: app/views/katello/api/v2/content_facet/show.json.rabl [hunk 3]
```diff
@@ -3,20 +3,22 @@ child :content_facet => :content_facet_attributes do
...
+        :content_view_version => version.version,
+        :content_view_version_id => version.id,
+        :content_view_version_latest => version.latest?
+      }
+    end
   end
 
   node :content_view_default? do |content_facet|
-    content_facet.single_content_view.default?
+    ::Foreman::Cast.to_bool(content_facet.single_content_view&.default?)
```
Comment (ekohl) [inline app/views/katello/api/v2/content_facet/show.json.rabl:17]: I'd keep it simple:
```suggestion
    content_facet.single_content_view&.default? || false
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Please squash the commits.

File: app/views/katello/api/v2/content_facet/show.json.rabl (hunk 2 above)
Comment (jeremylenz) [reply app/views/katello/api/v2/content_facet/show.json.rabl]: `content_view_default?` should return a Boolean; I wanted to ensure it's not `nil`. Switched to `::Foreman::Cast.to_bool`.

Comment (jeremylenz) [reply app/views/katello/api/v2/content_facet/show.json.rabl]: ...and now  to `|| false`

**Comment by ekohl** [comment]
Code looks ok, but can we please also cover this using tests so we don't regress in the future?

File: test/controllers/api/v2/hosts_controller_test.rb [hunk 4]
```diff
@@ -41,6 +41,13 @@ def test_with_content
     host_index_and_show(host)
   end
 
+  def test_no_cves
```
Comment (evgeni) [inline test/controllers/api/v2/hosts_controller_test.rb:44]:
```suggestion
  def test_no_content_view_environments
```

CVEs is kinda ambiguous in our context ;-)

**Comment by evgeni** [comment]
LGTM besides one wording nitpick ;-)
(didn't run it, but it reads correct)

**Comment by jeremylenz** [comment]
@evgeni updated :)

**Review by evgeni — APPROVED** [review]

**Review by parthaa — APPROVED** [review]
ack
